rtos/TARGET_CORTEX/rtx5/RTX/Source/rtx_kernel.c

```diff
@@ -93,8 +93,11 @@ osStatus_t svcRtxKernelInitialize (void) {
     return osError;
   }
 
-  // Initialize osRtxInfo
-  memset(&osRtxInfo.kernel, 0, sizeof(osRtxInfo) - offsetof(osRtxInfo_t, kernel));
+  // Initialize osRtxInfo, skipping all elements before the kernel struct,
+  // (os_id and version), which are set elsewhere
+  memset(&osRtxInfo + sizeof(osRtxInfo.os_id) + sizeof(osRtxInfo.version),
```

Comment:
Can you send this upstream CMSIS_5 repo?

With this change, any change to info structure would means changing this code (previously the offset was calculated using offsetof macro).

Reply:
I will send it upstream and see what they say.
I understand that reasonable people will disagree on whether this is easier to read/maintain than what was there.
My main motivation was to clean up the Coverity reports, which this change should do.
We can see what the CMSIS_5 people want to do with it.

Reply:
I do not yet see why Coverity complains (what is the issue there) - when this out of bounds access can happen?

Reply:
I am a bit concerned. Fixes for reasonable issues found by additional analysis tools (i.e. Coverity) are highly appreciated, for sure. In this case one can discuss which implementation is "easier to read". 

From maintenance point of view the former implementation is less vulnerable to structure changes than the latter one. Admittedly I assume changes to osRtxInfo_t to be made very carefully if needed at all. 

Thus I would like to understand the issues Coverity complains about. I do not understand why the latter implementation should be more safe than the former one. Does the warning correlate with `offsetof` macro?